conjure/src/main/java/com/palantir/conjure/cli/ConjureCli.java

```diff
@@ -58,61 +57,81 @@ public static void main(String[] args) {
         System.exit(prepareCommand().execute(args));
     }
 
+    public static void inProcessExecution(String[] args) {
+        ExceptionHandler exceptionHandler = new ExceptionHandler();
+        new CommandLine(new ConjureCli())
+                .setExecutionExceptionHandler(exceptionHandler)
+                .execute(args);
+        exceptionHandler.maybeRethrowException();
+    }
+
     @VisibleForTesting
     static CommandLine prepareCommand() {
         return new CommandLine(new ConjureCli()).setExecutionExceptionHandler(new ExceptionHandler());
     }
 
     public static final class ExceptionHandler implements IExecutionExceptionHandler {
+
+        private Optional<Exception> thrownException = Optional.empty();
+
         @Override
         public int handleExecutionException(Exception ex, CommandLine commandLine, ParseResult _parseResult)
                 throws Exception {
+
             if (ex == null) {
+                thrownException = Optional.ofNullable(ex);
                 return 0;
             }
             if (!(commandLine.getCommand() instanceof ConjureCliCommand)) {
+                thrownException = Optional.ofNullable(ex);
                 throw ex;
             }
             ConjureCliCommand cmd = commandLine.getCommand();
             if (cmd.isVerbose()) {
                 // If the command is verbose, print the full stack trace
+                thrownException = Optional.ofNullable(ex);
                 throw ex;
             }
             List<Throwable> chain = Throwables.getCausalChain(ex);
 
             // For known internal errors, change logging, as they're signaling errors on the input, not in the code
             if (chain.stream().noneMatch(ExceptionHandler::isConjureThrowable)) {
+                thrownException = Optional.ofNullable(ex);
                 throw ex;
             }
 
             // Unpack errors; we don't care about where in the code the error comes from: the issue is in the supplied
             // conjure code. The stack doesn't help.
-            chain.forEach(exception -> {
-                if (exception instanceof JsonMappingException) {
-                    handleJsonMappingException(commandLine.getErr(), (JsonMappingException) exception);
-                } else {
-                    commandLine.getErr().println(exception.getMessage());
-                }
-            });
+            String message = chain.stream()
+                    .map(exception -> (exception instanceof JsonMappingException)
+                            ? handleJsonMappingException((JsonMappingException) exception)
+                            : exception.getMessage())
+                    .collect(Collectors.joining("\n"));
+            commandLine.getErr().println(message);
```

Comment:
What happens to this when called in process? Does it still print out to stderr?

Reply:
Looks like there is a `setErr` we can use on `CommandLine` to change it.

Reply:
<img width="1694" alt="Screenshot 2024-03-04 at 2 54 22 PM" src="https://github.com/palantir/conjure/assets/25724655/a0116711-dc39-40ee-957b-893e816d73f6">
Yes, it does still log to stderr. the getErr() is the default standardError: https://github.com/remkop/picocli/blob/main/src/main/java/picocli/CommandLine.java#L1311C16-L1311C19